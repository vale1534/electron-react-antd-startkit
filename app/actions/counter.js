// @flow
import type {Dispatch, GetState} from '../types';
import * as defs from './defs';

export function increment() {
  return { type: defs.kIncrementCounter };
}

export function decrement() {
  return {
    type: defs.kDecrementCounter
  };
}

export function incrementAsync() {
  return {
    type: defs.kIncrementCounterAsync
  };
}

export function incrementIfOdd() {
  return (dispatch: Dispatch, getState: GetState) => {
    const { counter } = getState();
    if (counter % 2 === 0) return;
    dispatch(increment());
  };
}
