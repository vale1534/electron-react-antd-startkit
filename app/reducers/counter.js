// @flow
import * as defs from '../actions/defs';

export default function counter(state = 0, action) {
  switch (action.type) {
    case defs.kIncrementCounter:
      return state + 1;
    case defs.kDecrementCounter:
      return state - 1;
    default:
      return state;
  }
}
