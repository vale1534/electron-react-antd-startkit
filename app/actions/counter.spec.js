import * as actions from './counter';

describe('counter actions', () => {
  it('t01 - snapshot testing', () => {
    expect(actions.increment()).toMatchSnapshot();
    expect(actions.decrement()).toMatchSnapshot();
    expect(actions.incrementIfOdd()).toMatchSnapshot();
    expect(actions.incrementAsync()).toMatchSnapshot();
  });

  it('t02', () => {
    expect(1).toBe(1);
  });
});
