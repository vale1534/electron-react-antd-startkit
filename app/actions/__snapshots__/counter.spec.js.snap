// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter actions t01 - snapshot testing 1`] = `
Object {
  "type": "kIncrementCounter",
}
`;

exports[`counter actions t01 - snapshot testing 2`] = `
Object {
  "type": "kDecrementCounter",
}
`;

exports[`counter actions t01 - snapshot testing 3`] = `[Function]`;

exports[`counter actions t01 - snapshot testing 4`] = `
Object {
  "type": "kIncrementCounterAsync",
}
`;
